Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -1162,6 +1163,169 @@ WebIDL::ExceptionOr<GC::Ref<Node>> Node::clone_node(Document* document, bool sub
     return GC::Ref { *copy };
 }
 
+// https://dom.spec.whatwg.org/#move
+WebIDL::ExceptionOr<void> Node::move_node(Node& new_parent, Node* child)
+{
+    // 1. If newParent’s shadow-including root is not the same as node’s shadow-including root, then throw a "HierarchyRequestError" DOMException.
+    if (&new_parent.shadow_including_root() != &shadow_including_root())
```

Comment:
This is what _should_ be catching the cross document case in the spec as I understand it, but isn't currently